sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala

```diff
@@ -663,4 +664,18 @@ class StringFunctionsSuite extends QueryTest with SharedSparkSession {
         start = 7,
         stop = 47))
   }
+
+  test("SPARK-41780: INVALID_PARAMETER_VALUE - invalid parameters `regexp` in regexp_replace") {
+    checkError(
+      exception = intercept[SparkRuntimeException] {
+        sql("select regexp_replace('', '[a\\\\d]{0, 2}', 'x')").collect()
+      },
+      errorClass = "INVALID_PARAMETER_VALUE",
+      parameters = Map(
+        "parameter" -> "regexp",
+        "functionName" -> "`regexp_replace`",
+        "expected" -> "'[a\\\\d]{0, 2}'"
```

Comment:
Looks like the parameter name is misnamed in INVALID_PARAMETER_VALUE https://github.com/apache/spark/blob/f352f103ed512806abb3f642571a0c595b8b0509/core/src/main/resources/error/error-classes.json#L792 

It should probably called something like `actual` or `invalid` , it's anything but `expected`